targets/csharp-unityv2/make.js

```diff
@@ -0,0 +1,354 @@
+    copyFile(path.resolve(sourceDir, "Testing/Tests/Client/ClientEventTest.cs"), path.resolve(baseTestingOutputDir, "Tests/Client/ClientEventTest.cs"));
+
+    copyFile(path.resolve(sourceDir, "Settings/Client/PlayFabSettings.cs"), path.resolve(baseApiOutputDir, "Client/PlayFabSettings.cs"));
+    // Add the DemoScene to the clientSDK - TODO: A command line parameter that decides when to add it or not, TODO: GitIgnore the DemoScene folder?
+    // copyTree(path.resolve(sourceDir, "testing/DemoScene"), path.resolve(baseApiOutputDir, "DemoScene"));
+}
+
+exports.makeServerAPI = function (apis, sourceDir, apiOutputDir) {
+    apiOutputDir = path.resolve(apiOutputDir, "Source/PlayFabSDK");
+    console.log("  - Generating C-sharp Unity server SDK sample proj to\n  -> " + apiOutputDir);
+    var filteredApis = apis.filter(function(el){ return el.name != "Admin"; });
+    makeDatatypes(filteredApis, sourceDir, apiOutputDir);
+    for (var i in filteredApis) {
+        makeAPI(filteredApis[i], sourceDir, apiOutputDir);
+    }
+    copyFile(path.resolve(sourceDir, "Settings/Server/PlayFabSettings.cs"), path.resolve(apiOutputDir, "Server/PlayFabSettings.cs"));
+}
+
+exports.makeAdminAPI = function (apis, sourceDir, apiOutputDir) {
+    apiOutputDir = path.resolve(apiOutputDir, "Source/PlayFabSDK");
+    console.log("  - Generating C-sharp Unity server SDK sample proj to\n  -> " + apiOutputDir);
+
+    makeDatatypes(apis, sourceDir, apiOutputDir);
+    for (var i in apis) {
+        makeAPI(apis[i], sourceDir, apiOutputDir);
+    }
+    copyFile(path.resolve(sourceDir, "Settings/Admin/PlayFabSettings.cs"), path.resolve(apiOutputDir, "Admin/PlayFabSettings.cs"));
+}
+
+function makeEvent(api,sourceDir,apiOutputDir){
+    var templateDir = path.resolve(sourceDir, "templates");
+    var apiTemplate = ejs.compile(readFile(path.resolve(templateDir, "ApiEvent.cs.ejs")));
+    var apiLocals = {};
+    apiLocals.api = api;
+    var generatedApi = apiTemplate(apiLocals);
+    writeFile(path.resolve(apiOutputDir,api.name +  "/PlayFabEvents.cs"), generatedApi);
+}
+
+function getBaseTypeSyntax(datatype) {
+    if (datatype.name.toLowerCase().indexOf("result") > -1 || datatype.name.toLowerCase().indexOf("response") > -1)
+        return " : PlayFabResultCommon";
+    return "";
+}
+
+function makeDatatypes(apis, sourceDir, apiOutputDir) {
+    var templateDir = path.resolve(sourceDir, "templates");
+    
+    var modelTemplate = ejs.compile(readFile(path.resolve(templateDir, "Model.cs.ejs")));
+    var modelsTemplate = ejs.compile(readFile(path.resolve(templateDir, "Models.cs.ejs")));
+    var enumTemplate = ejs.compile(readFile(path.resolve(templateDir, "Enum.cs.ejs")));
+    
+    var makeDatatype = function (datatype) {
+        var modelLocals = {};
+        modelLocals.datatype = datatype;
+        modelLocals.getPropertyDef = getModelPropertyDef;
```

Comment:
You should fix ReSharper errors in your make.js before you submit this.  I've been trying to reduce the resharper issues, not grow them.